### PR TITLE
WT-8721 Use credentials from the configuration to create connection to S3

### PR DESCRIPTION
**Pull request by fionaochee**
* Use credentials from the`authToken` parameter in `S3CustomizeFileSystem` to initialize a connection to the bucket.
* Update python test to provide the credentials.

### CODE REVIEW

**Review by sulabhM — APPROVED** [review]

**Review by etienneptl — APPROVED** [review]
Nice one, LGTM!

File: ext/storage_sources/s3_store/s3_storage_source.cpp [hunk 1]
```diff
@@ -436,20 +436,34 @@ S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, con
...
     s3 = (S3_STORAGE *)storageSource;
 
+    /* We need to have a bucket to setup the file system. */
+    if (bucketName == NULL || strlen(bucketName) == 0) {
+        s3->log->LogVerboseErrorMessage("S3CustomizeFileSystem: bucket not specified.");
+        return (EINVAL);
+    }
+
+    /* Fail if there is no authentication provided. */
+    if (authToken == NULL || strlen(authToken) < 1) {
```
Comment (etienneptl) [inline ext/storage_sources/s3_store/s3_storage_source.cpp:446]: np: You check against `1` here and `0` line 440. We should be consistent.

File: ext/storage_sources/s3_store/s3_storage_source.cpp [hunk 2]
```diff
@@ -436,20 +436,34 @@ S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, con
...
+
     /* Extract the AWS access key ID and the AWS secret key from authToken. */
     int delimiter = std::string(authToken).find(',');
+    if (delimiter == std::string::npos) {
+        s3->log->LogVerboseErrorMessage("S3CustomizeFileSystem: authToken malformed.");
+        return (EINVAL);
+    }
     const std::string accessKeyId = std::string(authToken).substr(0, delimiter);
     const std::string secretKey = std::string(authToken).substr(delimiter + 1);
+    if (accessKeyId.size() == 0 || secretKey.size() == 0) {
```
Comment (etienneptl) [inline ext/storage_sources/s3_store/s3_storage_source.cpp:459]: np: Can we use [empty()](https://www.cplusplus.com/reference/string/string/empty/) instead?

File: test/suite/test_s3_store01.py [hunk 3]
```diff
@@ -50,7 +50,11 @@ class test_s3_store01(wttest.WiredTigerTestCase):
         bucket_name = "s3testext"
 
     # Auth token contains the AWS access key ID and the AWS secret key as comma-separated values.
-    auth_token = os.getenv('AWS_ACCESS_KEY_ID') + "," + os.getenv('AWS_SECRET_ACCESS_KEY')
+    access_key = os.getenv('AWS_ACCESS_KEY_ID')
+    secret_key = os.getenv('AWS_SECRET_ACCESS_KEY')
+    auth_token = None
+    if access_key and secret_key:
```
Comment (etienneptl) [inline test/suite/test_s3_store01.py:56]: Does the test fail otherwise?

File: ext/storage_sources/s3_store/s3_storage_source.cpp (hunk 2 above)
Comment (sulabhM) [reply ext/storage_sources/s3_store/s3_storage_source.cpp]: done.

**Review by etienneptl — APPROVED** [review]
LGTM

**Review by sulabhM — APPROVED** [review]

**Pull request merged**